src/zenml/zen_server/pipeline_deployment/workload_manager_interface.py

```diff
@@ -0,0 +1,109 @@
+"""Workload manager interface definition."""
+from abc import ABC, abstractmethod
+from typing import Dict, List, Optional, Type
+from uuid import UUID
+
+from zenml.utils import source_utils
+from zenml.zen_server.utils import server_config
+
+
+class WorkloadManagerInterface(ABC):
+    """Workload manager interface."""
+
+    @abstractmethod
+    def __init__(self, workload_id: UUID) -> None:
```

Comment:
Yep I guess this is meant to be more generic than just to run pipelines from the dashboard, as this can essentially be used to build generic Docker images or run other stuff as well. I'll add it to the docstring if that works for you, we can also revert this to a deployment ID if you think that makes it less confusing for now.

Reply:
Nah, if it's meant to be generic, then it makes sense to keep it `workload_id`.